### DISCUSSION TITLE
Intermediate Parameters User Auditing

### DISCUSSION THREAD

**Discussion by justincbeard** [argoproj/argo-workflows#9539] [Q&A] [answered]
I'm trying to find a way to log the username or email of a user who interacts with a workflow via intermediate parameters.  I am creating a workflow for terraform deployments that has an intermediate step for the approval of the generated plans.  For auditing purposes it would be ideal to record who interacted with this intermediate suspend step.  

Is there anyway to make this happen?

**Comment by terrytangyuan** [comment, marked as answer]
It's not supported yet. It should be straightforward to add a log somewhere if anyone wants to contribute.

**Reply by tooptoop4** [reply]
@terrytangyuan I am interested, can u give any pointer on what file to change and how to derive the value?

**Comment by praveenjindal62** [comment]
Do we have approver information logged anywhere else ? For example in status section of given workflow ? 

This would be nice to have to log user information, until then if that information is available somewhere, we can definitely fetch that info and log manually.
